lib/extractor/image.ts

```diff
@@ -1,25 +1,145 @@
-export function getImageParts(imageWithTag: string): {
-  imageName: string;
-  imageTag: string;
-  imageDigest: string;
-} {
-  // we're matching pattern: <registry:port_number>(optional)/<image_name>(mandatory):<image_tag>(optional)@<tag_identifier>(optional)
-  // extracted from https://github.com/docker/distribution/blob/master/reference/regexp.go
-  const regex =
-    /^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?\/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:\/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][A-Fa-f0-9]{32,}))?$/gi;
-  const groups = regex.exec(imageWithTag);
-
-  if (!groups) {
-    throw new Error("Image with tag is malformed, cannot extract valid parts!");
+export { ImageName, ImageDigest, DigestType };
+class ImageName {
```

Comment:
Nit: may need to be public, if `snyk-monitor` would have to provide an `ImageName` passed through the plugin options